cli-output/src/cli_output.rs

```diff
@@ -610,25 +610,63 @@ fn show_votes_and_credits(
         return Ok(());
     }
 
-    writeln!(f, "Recent Votes:")?;
-    for vote in votes {
-        writeln!(f, "- slot: {}", vote.slot)?;
-        writeln!(f, "  confirmation count: {}", vote.confirmation_count)?;
-    }
-    writeln!(f, "Epoch Voting History:")?;
+    // Existence of this should guarantee the occurrence of vote truncation
+    let newest_history_entry = epoch_voting_history.iter().rev().next();
+
     writeln!(
         f,
-        "* missed credits include slots unavailable to vote on due to delinquent leaders",
+        "{} Votes (using {}/{} entries):",
```

Comment:
The subtle thing here is that 29/31 is possible even if vote truncation started because fork switch. So, the `Recent`/`All` title switching and this number indicator isn't strictly-speaking redundant information.